src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/ContentService.cs

```diff
@@ -155,14 +158,14 @@ public class ContentService : IContentService
                     {
                         var (release, sectionToRemove) = tuple;
 
-                        // detach DataBlocks before removing the ContentSection and its ContentBlocks
-                        var dataBlocks = sectionToRemove
+                        await sectionToRemove
                             .Content
-                            .OfType<DataBlock>()
-                            .ToList();
-
-                        dataBlocks.ForEach(dataBlock =>
-                            RemoveContentBlockFromContentSection(sectionToRemove, dataBlock, false));
+                            .ToList()
+                            .ToAsyncEnumerable()
+                            .ForEachAwaitAsync(async contentBlock =>
+                            {
+                                await _contentBlockService.DeleteContentBlockAndReorder(contentBlock.Id, reorder: false);
```

Comment:
Would suggest we get rid of the `reorder` flag. It's a bit of code smell to have flags that modify behaviour, especially if they do the exact opposite of what the method says.

Another approach would be to define a new `DeleteSectionContentBlocks(Guid contentSectionId)` method in `ContentBlockService` which handles deleting all of a section's content blocks for us without trying to reorder anything afterwards.

Reply:
I've removed the reorder bool and created a `DeleteSectionContentBlocks` method.